lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -623,6 +644,7 @@ namespace Js
             });
         }
 
+        ModuleNamespace::GetModuleNamespace(this);
         Js::AutoDynamicCodeReference dynamicFunctionReference(scriptContext);
```

Comment:
yeah. The logic gets a bit messy to do the resolve time initialization with the slot setup etc. One instance per namespace and I don't think the impact would be big, and the code is overall cleaner. We can change that if it becomes an issue.

---
In reply to: [74169052](https://github.com/Microsoft/ChakraCore/pull/1401#discussion_r74169052) [](ancestors = 74169052)